src/invocation/InvocationService.ts

```diff
@@ -21,12 +23,16 @@ class InvocationService {
         this.smartRoutingEnabled = hazelcastClient.getConfig().networkConfig.smartRouting;
     }
 
-    invokeOnConnection(connection: ClientConnection, request: ClientMessage): Q.Promise<ClientMessage> {
+    invokeOnConnection(connection: ClientConnection, request: ClientMessage, handler: (...args: any[]) => any = null)
```

Comment:
Passing handler function argument here looks really ugly to me. 

@gAmUssA What do you think?